docs/plugins/development.md

```diff
@@ -0,0 +1,243 @@
+* `url_slug` - Base path to use for plugin URLs (optional). If not specified, the project's `name` will be used.
+* `required_settings`: A list of configuration parameters that **must** be defined by the user
+* `default_settings`: A dictionary of configuration parameter names and their default values
+* `min_version`: Minimum version of NetBox with which the plugin is compatible
+* `max_version`: Maximum version of NetBox with which the plugin is compatible
+* `middleware`: A list of middleware classes to append after NetBox's build-in middleware.
+* `caching_config`: Plugin-specific cache configuration
+
+## Database Models
+
+Plugins can define their own Django models to record user data. A model is a Python representation of a database table. Model instances can be created, manipulated, and deleted using the [Django ORM](https://docs.djangoproject.com/en/stable/topics/db/). Models are typically defined within a plugin's `models.py` file, though this is not a strict requirement.
+
+Below is a simple example `models.py` file showing a model with two character fields:
+
+```python
+from django.db import models
+
+class Animal(models.Model):
+    name = models.CharField(max_length=50)
+    sound = models.CharField(max_length=50)
+
+    def __str__(self):
+        return self.name
+```
+
+Once you have defined the model(s) for your plugin, you'll need to create the necessary database schema migrations as well. This can be done using the Django `makemigrations` management command:
+
+```no-highlight
+$ ./manage.py makemigrations netbox_animal_sounds 
+Migrations for 'netbox_animal_sounds':
+  /home/jstretch/animal_sounds/netbox_animal_sounds/migrations/0001_initial.py
+    - Create model Animal
+```
+
+Once the migration has been created, we can apply it locally with the `migrate` command:
+
+```no-highlight
+$ ./manage.py migrate netbox_animal_sounds
+Operations to perform:
+  Apply all migrations: netbox_animal_sounds
+Running migrations:
+  Applying netbox_animal_sounds.0001_initial... OK
+```
+
+For more information on database migrations, see the [Django documentation](https://docs.djangoproject.com/en/stable/topics/migrations/).
+
+### Using the Django Admin Interface
+
+Plugins can optionally expose their models via Django's built-in [administrative interface](https://docs.djangoproject.com/en/stable/ref/contrib/admin/). This can greatly improve troubleshooting ability, particularly during development. An example `admin.py` file for the above model is shown below:
+
+```python
+from django.contrib import admin
+from .models import Animal
+
+@admin.register(Animal)
+class AnimalAdmin(admin.ModelAdmin):
+    list_display = ('name', 'sound')
+``` 
+
+This will display the plugin and its model in the admin UI. Staff users can create, change, and delete model instances via the admin UI without needing to create a custom view.
+
+![NetBox plugin in the admin UI](../media/plugins/plugin_admin_ui.png)
+
+## Views
```

Comment:
It might be worth linking to our existing development docs on the base view classes NetBox uses. That section needs some work, but we can do that over time.

Reply:
I'm going to take a hard line against providing any sort of API for our internal view functions. They are for internal use only and indefinitely subject to change. Plugin authors need to either create their own views or accept that integration might randomly break in any release.